addons/nuxeo-template-rendering/nuxeo-template-rendering-core/src/main/java/org/nuxeo/template/serializer/executors/XMLSerializer.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Benjamin JALON
+ *     Thierry Delprat
+ *
+ */
+
+package org.nuxeo.template.serializer.executors;
+
+import static org.nuxeo.template.api.InputType.StringValue;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.dom4j.Document;
+import org.dom4j.DocumentException;
+import org.dom4j.DocumentFactory;
+import org.dom4j.DocumentHelper;
+import org.dom4j.Element;
+import org.dom4j.Namespace;
+import org.dom4j.QName;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.template.api.InputType;
+import org.nuxeo.template.api.TemplateInput;
+
+/**
+ * {@link TemplateInput} parameters are stored in the {@link DocumentModel} as a single String Property via XML
+ * Serialization. This class contains the Serialization/Deserialization logic.
+ *
+ * @author Tiry (tdelprat@nuxeo.com)
+ * @author bjalon (bjalon@qastia.com)
+ */
+public class XMLSerializer implements Serializer {
+
+    public static final String XML_NAMESPACE = "http://www.nuxeo.org/DocumentTemplate";
+    public static final String XML_NAMESPACE_PREFIX = "nxdt";
+    public static final Namespace ns = new Namespace(XML_NAMESPACE_PREFIX, XML_NAMESPACE);
+    public static final QName fieldsTag = DocumentFactory.getInstance().createQName("templateParams", ns);
+    public static final QName fieldTag = DocumentFactory.getInstance().createQName("field", ns);
+    public static final String DATE_FORMAT = "yyyy-MM-dd HH:mm:ss.SSS";
+
+    protected static final Log log = LogFactory.getLog(XMLSerializer.class);
```

Comment:
The `log` is always `private`, we don't want any inheritance.

Reply:
Ok. It seems in many place protected is used... I fixed just in my place... :)